internal/test/morerequire/morerequire.go

```diff
@@ -89,3 +95,43 @@ func RequireChdirIntoTemp(t *testing.T) func() {
 		removeTempDir()
 	}
 }
+
+var (
+	// fakeEnvoySrc is a test source file used for capturing arguments and signals. If it receives an argument "exit=N",
+	// where N is a code number, the script exits with that status. Otherwise it sleeps until a signal interrupts it.
+	//go:embed testdata/fake_envoy.go
+	fakeEnvoySrc []byte
+)
+
+// requireBuildFakeEnvoy builds a fake envoy binary and returns its contents.
+func requireBuildFakeEnvoy(t *testing.T) []byte {
+	goBin := requireGoBin(t)
+	tempDir, deleteTempDir := RequireNewTempDir(t)
+	defer deleteTempDir()
+
+	name := "envoy"
+	src := name + ".go"
+	require.NoError(t, os.WriteFile(filepath.Join(tempDir, src), fakeEnvoySrc, 0600))
+	cmd := exec.Command(goBin, "build", "-o", name, src) //nolint:gosec
```

Comment:
only tricky part is we invoke the compiler. This will fail if someone someone is trying to run tests without tools installed (if that's even possible). Possibly some IDE may not like this, but goland does. 

This is similar to what golang does here, just a bit simpler https://github.com/golang/go/blob/master/src/os/exec/lp_windows_test.go#L523